src/packages/auth/routes.ts

```diff
@@ -47,6 +48,61 @@ export default (app: Express) => {
     res.json(response);
   });
 
+  /**
+   * @swagger
+   * /auth/validateSSO:
+   *   get:
+   *     parameters:
+   *       - in: cookie
+   *         name: AUTH_SSO_TOKEN_NAME
+   *         schema:
+   *           type: string
+   *         description: SSO token cookie that is named according to the gateway environment variable
+   *     produces:
+   *       - application/json
+   *     responses:
+   *       200:
+   *         description: AuthResponse
+   *     summary: Validates a user's SSO token against external auth providers
+   *     tags:
+   *       - Auth
+   */
+  app.get('/auth/validateSSO', loginLimiter, async (req, res) => {
+    const { token, success, message, userId, redirectURL } = await auth.validate(req);
+    const resp = {
+      message,
+      redirectURL,
+      success,
+      token,
+      userId,
+    };
+    res.json(resp);
+  });
```

Comment:
The idea was to leave the current `/auth/login` and `/auth/session` endpoints unchanged and create new SSO endpoints instead of overloading the existing ones.

For example, the `/auth/validateSSO` endpoint expects cookies to be forwarded in requests, v.s. the `/auth/session` endpoint expects an `Authorization: Bearer ...` header in requests. We could combine the two auth flows into one endpoint, but then more logic would have to live in the handler, instead of letting the client decide exactly which endpoints / auth to use.